ls1intum/Artemis
# [Feature] second correction for exam file upload exercises

### Checklist

### Motivation and Context


In [#2833](https://github.com/ls1intum/Artemis/pull/2833) we added file upload exercises to the exam mode. There we didn't yet add the second correction.

### Description
In this PR we will add the second correction round to file-upload exercises in the exam mode, and also adapt the file-upload-assessment dashboard

assessment-dashboard:
![image](https://user-images.githubusercontent.com/33342534/107696891-cf21a800-6cb2-11eb-8119-b462cf18410c.png)

exercise-assessment-dashboard:
![image](https://user-images.githubusercontent.com/33342534/107696949-de085a80-6cb2-11eb-9468-dd784815f610.png)

file-upload-assessment-dashboard:
![image](https://user-images.githubusercontent.com/33342534/107696822-b618f700-6cb2-11eb-960d-c06137a7da8a.png)

### Steps for Testing
1. create a new exam with a file upload exercise
2. participate in the exam with a few students
3. assess some of the submissions in the first correction round with a tutor
4. Now check with an instructor in the file-upload-assessment-dashboard (exam -> exercise-groups -> Submissions) if you can see all of the assessed submissions, and if the correct results/scores are loaded when you use "open assessment"
5. test the "new assessment" function with one of the remaining submissions
6. Now, enable the second correction
7. use a different tutor account to assess in the second correction round. Make sure that the results of the first assessment are loaded.
8. After some assessments again double-check in the instructor's file-upload-assessment-dashboard if everything worked fine 

### Test Coverage



file-upload-assessment-dashboard: > 92.54%
